cmd/elastic.sh

```diff
@@ -3,10 +3,25 @@ set -e;
 
 function elastic_schema_drop(){ compose_run 'schema' node scripts/drop_index "$@" || true; }
 function elastic_schema_create(){ compose_run 'schema' ./bin/create_index; }
+
+# perform pre-start checks and start the elasticsearch container
 function elastic_start(){
   mkdir -p $DATA_DIR/elasticsearch
   # attemp to set proper permissions if running as root
   chown $DOCKER_USER $DATA_DIR/elasticsearch 2>/dev/null || true
+
+  # record the owner of the Elasticsearch directory
+  elasticsearch_owner_uid=$(stat --format '%u' $DATA_DIR/elasticsearch)
```

Comment:
Still not working for me...

```bash
stat --format '%u' /data/pelias/docker/projects/portland-metro/elasticsearch
stat: illegal option -- -
usage: stat [-FlLnqrsx] [-f format] [-t timefmt] [file ...]
```

I think you might need to use the g* version:

```bash
gstat --format '%u' /data/pelias/docker/projects/portland-metro/elasticsearch
501
```

There is already a process for handling GNU versions of utils under OSX in https://github.com/pelias/docker/blob/master/pelias#L10

Reply:
Good catch yet again. It turns out GNU coreutils and BSD coreutils do not have compatible flags here.

My `.bashrc` adds the [homebrew coreutils directory to my $PATH](https://github.com/orangejulius/dotfiles/blob/master/bashrc_mac#L18-L20). Definitely convenient, but we can't expect everyone to do that.

I've added an environment variable for the `stat` command with a similar pattern. It should _really_ work this time.

Reply:
Hmm.. I still don't think this is working as expected...

I've never had a problem with permissions on my laptop and now I would get a failure when trying to start `elasticsearch`, which would also probably be the case for all other Mac users too?

I don't have anything special set up for my environment, I decided to avoid aliases etc in order to have a 'stock standard' Mac dev environment which was closest to what our users would be running.

```bash
➜  portland-metro git:(check-data-dir-permissions) ✗ pelias elastic start
user 1000 cannot access elasticsearch directory at /data/pelias/docker/projects/portland-metro
please run 'sudo chown 1000 /data/pelias/docker/projects/portland-metro/elasticsearch'
```
```bash
➜  portland-metro git:(check-data-dir-permissions) ✗ ls -lah /data/pelias/docker/projects/portland-metro
total 0
drwxr-xr-x  14 peter  staff   448B Mar  4 16:33 .
drwxr-xr-x   3 peter  staff    96B Feb  8 08:14 ..
drwxr-xr-x   2 peter  staff    64B Feb  8 08:18 blacklist
drwxr-xr-x   3 peter  staff    96B Feb  8 08:18 csv
drwxr-xr-x   3 peter  staff    96B Mar  4 16:33 elasticsearch
drwxr-xr-x   3 peter  staff    96B Mar  4 16:32 es-bak
drwxr-xr-x  20 peter  staff   640B Feb  8 08:41 interpolation
drwxr-xr-x   4 peter  staff   128B Feb  8 08:18 openaddresses
drwxr-xr-x   3 peter  staff    96B Feb  8 08:18 openstreetmap
drwxr-xr-x   4 peter  staff   128B Feb  8 08:23 placeholder
drwxr-xr-x   3 peter  staff    96B Feb  8 08:22 polylines
drwxr-xr-x   4 peter  staff   128B Feb  8 08:18 tiger
drwxr-xr-x  16 peter  staff   512B Feb  8 08:20 transit
drwxr-xr-x   5 peter  staff   160B Feb  8 08:19 whosonfirst
```
```bash
➜  portland-metro git:(check-data-dir-permissions) ✗ ls -lah /data/pelias/docker/projects/portland-metro/elasticsearch
total 0
drwxr-xr-x   3 peter  staff    96B Mar  4 16:33 .
drwxr-xr-x  14 peter  staff   448B Mar  4 16:33 ..
drwxr-xr-x   3 peter  staff    96B Mar  4 16:33 nodes
```
```bash
➜  portland-metro git:(check-data-dir-permissions) ✗ whoami
peter
```
```bash
➜  portland-metro git:(check-data-dir-permissions) ✗ groups
staff operator everyone localaccounts _appserverusr admin _appserveradm _lpadmin com.apple.sharepoint.group.1 _appstore _lpoperator _developer _analyticsusers com.apple.access_ftp com.apple.access_screensharing com.apple.access_ssh
```
```bash
➜  portland-metro git:(check-data-dir-permissions) ✗ gstat --format '%u' /data/pelias/docker/projects/portland-metro/elasticsearch
501
```

Reply:
One solution is to change the ID in https://github.com/pelias/docker/blob/master/projects/portland-metro/.env#L3 as per the readme.
This seems like an additional step for Mac users which will deter them from adopting Pelias, I'd really love it to 'just work first time'.

Reply:
It seems as though the `chown` command is failing due to a permissions error, the messages are being silenced:

```bash
➜  portland-metro git:(check-data-dir-permissions) ✗ chown 1000 /data/pelias/docker/projects/portland-metro/elasticsearch
chown: /data/pelias/docker/projects/portland-metro/elasticsearch: Operation not permitted
```
```bash
➜  portland-metro git:(check-data-dir-permissions) ✗ echo $?
1
```